### PR TITLE
[pten]  remove deprecated fluid op kernel for pten 

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

New features
### PR changes

OPs
### Describe

[pten] update cmake file to remove fluid kernel

- update cmake file for better generation of  "USE_OP" in `pybind.h` to support operator without fluid cuda/cpu kernel
- add add pten declaration.h to where pybind.h used
- update dygraph op generator
- fit eager legacy
- update kernel choosing, to choose pten cpu kernel first if kernel of specific device not found

### CODE REVIEW

**Review by XieYunshen — DISMISSED** [review]

File: cmake/operators.cmake [hunk 1]
```diff
@@ -1,6 +1,29 @@
 # CMake file `unity_build` is used to handle Unity Build compilation.
 include(unity_build)
 set(PART_CUDA_KERNEL_FILES)
+
+function(find_register FILENAME PATTERN OUTPUT)
+# find the op_name of REGISTER_OPERATOR(op_name, ...), REGISTER_OP_CPU_KERNEL(op_name, ...) , etc.
+# set op_name to OUTPUT
+set(options "")
```
Comment (chenwhql) [inline cmake/operators.cmake]: 这里是不是需要缩进

**Review by chenwhql — DISMISSED** [review]

File: cmake/operators.cmake (hunk 1 above)
Comment (zhiqiu) [reply cmake/operators.cmake]: done

**Review by XieYunshen — DISMISSED** [review]

**Review by chenwhql — DISMISSED** [review]

**Review by phlrain — DISMISSED** [review]

**Review by Aurelius84 — DISMISSED** [review]
LGTM

**Review by Ligoml — DISMISSED** [review]

**Review by chenwhql — DISMISSED** [review]

**Review by Ligoml — DISMISSED** [review]

**Review by jzhang533 — DISMISSED** [review]
LGTM

**Review by XieYunshen — DISMISSED** [review]

**Review by Aurelius84 — DISMISSED** [review]

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by raindrops2sea — DISMISSED** [review]

**Review by XieYunshen — DISMISSED** [review]

**Review by chenwhql — DISMISSED** [review]

**Review by chenwhql — DISMISSED** [review]

**Review by chenwhql — DISMISSED** [review]

**Review by chenwhql — DISMISSED** [review]

**Review by XieYunshen — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by Aurelius84 — APPROVED** [review]

**Review by Aurelius84 — APPROVED** [review]

**Review by Aurelius84 — APPROVED** [review]

**Review by Ligoml — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]

**Review by phlrain — APPROVED** [review]

**Pull request merged**